stream-api/src/main/java/com/bobocode/StreamTask_Collecting.java

```diff
@@ -14,15 +16,53 @@ public static void main(String[] args) {
         List<Account> accounts = TestDataProvider.generateAccountList();
 
         // 1 - Find a set of account owner's names(first name) that have a length <= 4
+        accounts.stream()
+                .filter(a -> a.getFirstName().length() <= 4)
+                .collect(Collectors.toSet())
+                .forEach(a -> System.out.println(a.getFirstName()));
+
+        System.out.println("--------------------------------");
 
         // 2 - Group all accounts by its email providers. (You can consider as an email provider
         // everything that goes after @ sign
 
+        accounts.stream()
+                .collect(Collectors.groupingBy(a -> a.getEmail().split("@")[1]))
+                .entrySet()
+                .stream()
+                .forEach(System.out::println);
+
+        System.out.println("--------------------------------");
+
         // 3 - Find the richest guy
 
+        Optional<Account> theRichestGuy = accounts.stream()
+                .collect(Collectors.maxBy(Comparator.comparing(Account::getBalance)));
+
+        System.out.println(theRichestGuy);
+
+        Optional<Account> theRichestGuy2 = accounts.stream()
+                .max(Comparator.comparing(Account::getBalance));
+
+        System.out.println(theRichestGuy2);
+
+        System.out.println("--------------------------------");
+
         // 4 - Find the richest guy for each email provider
 
+        accounts.stream()
+                .collect(Collectors.groupingBy(a -> a.getEmail().split("@")[1],
+                        Collectors.maxBy(Comparator.comparing(Account::getBalance))))
```

Comment:
it's easier to read when you use _static import_

Reply:
Totaly, but I left it for better understanding and remembering the topic.

Reply:
Okay